lib/daru/vector.rb

```diff
@@ -238,9 +238,15 @@ def [](*indexes)
           indexes = indexes.map { |e| named_index_for(e) }
         end
 
-        Daru::Vector.new(
-          indexes.map { |loc| @data[@index[loc]] }, 
-          name: @name, index: indexes, dtype: @dtype)
+        begin
+          Daru::Vector.new(
+            indexes.map { |loc| @data[@index[loc]] },
+            name: @name, index: indexes, dtype: @dtype)
+        rescue NoMethodError
+          raise IndexError, "Specified index #{pos.inspect} does not exist."
+        rescue TypeError
+          raise IndexError, "One or more of specified indices #{pos.inspect} do not exist."
```

Comment:
I'm not saying we should let `[:x, :y]` to fail. I'm saying that say you have a Vector `vector` with indexes `:a, :b, :c and :y`:
``` ruby
vector = Daru::Vector.new([1,2,3,4], index: [:a, :b, :c, :y])
```

Now say I wrongly say I want indexes `:x` and `:y`, I should say it like this:
``` ruby
vector[:x, :y]
```

And NOT like this:
``` ruby
vector[[:x, :y]] # notice Array inside #[] method
```

Therefore, the second way (i.e. `[:x, :y]`) should raise a `TypeError` because the `#[]` method is not supposed to accept Arrays as inputs.